mflowgen/Tile_PE/constraints/common.tcl

```diff
@@ -0,0 +1,144 @@
+set_input_delay -clock ${clock_name} ${i_delay} [all_inputs]
+# Pass through should have no input delay
+set_input_delay -clock ${clock_name} 0 clk_pass_through
+set_input_delay -clock ${clock_name} 0 stall
+set_input_delay -clock ${clock_name} 0 config_config_data*
+set_input_delay -clock ${clock_name} 0 config_config_addr*
+set_input_delay -clock ${clock_name} 0 config_read*
+set_input_delay -clock ${clock_name} 0 config_write*
+
+# Constrain OUTPUTS
+# set_output_delay constraints for output ports
+set o_delay [expr 0.0 * ${dc_clock_period}]
+set_output_delay -clock ${clock_name} ${o_delay} [all_outputs]
+
+# Set timing on pass through clock
+# Set clock min delay and max delay
+set clock_max_delay 0.05
+set_min_delay -from clk_pass_through -to clk*out 0
+set_max_delay -from clk_pass_through -to clk*out ${clock_max_delay}
+
+# Min and max delay a little more than our clock
+#set min_w_in [expr ${clock_max_delay} + ${i_delay}]
+set min_w_in ${clock_max_delay}
+set_min_delay -to config_out_config_addr* ${min_w_in}
+set_min_delay -to config_out_config_data* ${min_w_in}
+set_min_delay -to config_out_read* ${min_w_in}
+set_min_delay -to config_out_write* ${min_w_in}
+set_min_delay -to stall_out* ${min_w_in}
+
+# Pass through (not clock) timing margin
+set alt_passthru_margin 0.03
+set alt_passthru_max [expr ${min_w_in} + ${alt_passthru_margin}]
+set_max_delay -to config_out_config_addr* ${alt_passthru_max}
+set_max_delay -to config_out_config_data* ${alt_passthru_max}
+set_max_delay -to config_out_read* ${alt_passthru_max}
+set_max_delay -to config_out_write* ${alt_passthru_max}
+set_max_delay -to stall_out* ${alt_passthru_max}
+
+# 5fF approx load
+set mark_approx_cap 0.005
+set_load ${mark_approx_cap} config_out_config_addr*
+set_load ${mark_approx_cap} config_out_config_data*
+set_load ${mark_approx_cap} config_out_read* 
+set_load ${mark_approx_cap} config_out_write*
+set_load ${mark_approx_cap} stall_out*
+set_load ${mark_approx_cap} clk*out
+
+# Set max transition on these outputs as well
+set max_trans_passthru .05
+set_max_transition ${max_trans_passthru} config_out_config_addr*
+set_max_transition ${max_trans_passthru} config_out_config_data*
+set_max_transition ${max_trans_passthru} config_out_read* 
+set_max_transition ${max_trans_passthru} config_out_write*
+set_max_transition ${max_trans_passthru} stall_out*
+set_max_transition ${max_trans_passthru} clk*out
+
+# Set input transition to match the max transition on outputs
+set_input_transition ${max_trans_passthru} clk_pass_through
+set_input_transition ${max_trans_passthru} stall
+set_input_transition ${max_trans_passthru} config_config_data*
+set_input_transition ${max_trans_passthru} config_config_addr*
+set_input_transition ${max_trans_passthru} config_read*
+set_input_transition ${max_trans_passthru} config_write*
+#
+
+set read_config_data_timing 0.300
+set_max_delay -from config_config_addr* -to read_config_data ${read_config_data_timing}
```

Comment:
Could you get rid of or significantly loosen this constraint? It's not necessary to constrain the path from config_addr to read_config_data output this tightly. The only read_config_data path that need to be constrainted is read_config_data_in -> read_config_data (the pass through). 

I believe Max got rid of this in the mem tile constraints.